lib/material/material_gui.flow

```diff
@@ -1468,4 +1472,200 @@ MOverlayT(margin : Transform<double>, radius : Transform<double>, style : Transf
+
+		needsMore = map(requiredWidths, \w -> w > (availableWidth / i2d(itemCount)));
+		withEnoughSpace = mapi(requiredWidths, \i, w -> if (!needsMore[i]) w else 0.0);
+		needMoreSpace = mapi(requiredWidths, \i, w -> if (needsMore[i]) w else 0.0);
+
+		spaceUsedByEnough = fold(withEnoughSpace, 0.0, \acc, w -> acc + w);
+		needMoreCount = fold(needsMore, 0, \acc, need -> if (need) acc + 1 else acc);
+		enoughSpaceCount = itemCount - needMoreCount;
+        
+		if (needMoreCount == 0) {
+			remainingSpace = availableWidth - spaceUsedByEnough;
+			extraPerCell = remainingSpace / i2d(itemCount);
+			map(requiredWidths, \w -> max(w + extraPerCell, minCellWidth));
+		} else {
+			remainingSpace = availableWidth - spaceUsedByEnough;
+			equalShare = remainingSpace / i2d(needMoreCount);
+			mapi(requiredWidths, \i, w ->
+				if (needsMore[i]) max(equalShare, minCellWidth)
+				else max(w, minCellWidth)
+			);
+		}
+	}
+
+	cellWidths = calculateCellWidths;
+	lastSelected = make(None());
+	buttonsFocus = generate(0, length(items), \__ -> make(false));
+
+	makeButton = \i, item, isInteractive -> {
+		content = item.first;
+		states = item.second;
+		buttonState = if (isInteractive) states.first else [];
+		isSelected = states.second;
+
+		title = if (iconsOnly) "" else eitherFn(content.first, \text -> text.text, \ -> "");
+		textStyle = eitherFn(content.first, \text -> text.style, \ -> []);
+		iconName = eitherFn(content.second, \icon -> icon.name, \ -> "");
+		iconStyle = eitherFn(content.second, \icon -> icon.style, \ -> []);
+		accessibilityLabel = if (title != "") title else iconName;
+		buttonRole = if (multiSelect) "checkbox" else "radio";
+
+		cellWidth = elementAt(cellWidths, i, totalWidth / i2d(length(items)));
+        isFirst = i == 0;
+        isLast = i == length(items) - 1;
+		createIcon = \icon -> MIcon(icon, concat([defaultBgColor], iconStyle));
+        createCheckIcon = MIcon("check", concat(iconStyle, [MIconSize(20.0), defaultBgColor])) |> MBorderRight(8.);
+
+        if (isInteractive) MComponent(
+			concat(
+				[
+					MOnClick(\ -> {
+						if (!multiSelect) {
+							current = getValue(lastSelected);
+							switch (current) {
+								None(): {}
+								Some(prev): {
+									if (prev != isSelected) {
+										nextDistinct(prev, false);
+									}
+								}
+							}
+							if (!getValue(isSelected)) {
+								nextDistinct(lastSelected, Some(isSelected));
+							}
+						}
+						nextDistinct(isSelected, !getValue(isSelected));
+					}),
+					AccessRole(buttonRole),
+					MMobileForceHover(),
+					MButtonTitle(const(accessibilityLabel)),
+					FAccessAttribute("aria-checked", fselect(isSelected, FLift(\sel -> if (sel) "true" else "false"))),
+					MOnFocus(\f -> nextDistinct(buttonsFocus[i], f))
+				],
+				buttonState
+			),
+			\focus -> {
+				MSelect3(
+					isSelected,
+					focus.hover,
+					buttonsFocus[i],
+					\selected, hovered, focused ->
+						MFrameCustom(
```

Comment:
It's better to use MFrameCustomT, so content won't rerender on each selected/hovered/focused change